lib/matplotlib/quiver.py

```diff
@@ -952,35 +954,35 @@ def __init__(self, ax, *args,
 
     def _find_tails(self, mag, rounding=True, half=5, full=10, flag=50):
         """
-        Find how many of each of the tail pieces is necessary.  Flag
-        specifies the increment for a flag, barb for a full barb, and half for
-        half a barb. Mag should be the magnitude of a vector (i.e., >= 0).
-
-        This returns a tuple of:
+        Find how many of each of the tail pieces is necessary.
 
-            (*number of flags*, *number of barbs*, *half_flag*, *empty_flag*)
+        Parameters
+        ----------
+        mag : array-like
+            Vector magnitudes; must be non-negative.
+        rounding : bool, default: True
+            Whether to round or to truncate to the nearest half-bard.
+        half, full, flag : float, defaults: 5, 10, 50
+            Increments for a half-barb, a barb, and a flag.
 
-        The bool *half_flag* indicates whether half of a barb is needed,
-        since there should only ever be one half on a given
-        barb. *empty_flag* flag is an array of flags to easily tell if
-        a barb is empty (too low to plot any barbs/flags.
+        Returns
+        -------
+        n_flags, n_barbs : int array
+            For each entry in *mag*, the number of flags and barbs.
+        half_flag : bool array
+            For each entry in *mag*, whether a half-barb is needed.
+        empty_flag : bool array
+            For each entry in *mag*, whether nothing is drawn.
         """
-
         # If rounding, round to the nearest multiple of half, the smallest
         # increment
         if rounding:
-            mag = half * (mag / half + 0.5).astype(int)
-
-        num_flags = np.floor(mag / flag).astype(int)
-        mag = mag % flag
-
-        num_barb = np.floor(mag / full).astype(int)
-        mag = mag % full
-
+            mag = half * np.around(mag / half)
+        n_flags, mag = divmod(mag, flag)
```

Comment:
AFAICS this would not work for non-array inputs.

Reply:
Indeed.  However we only ever call this with actual arrays, so I just adjusted the docstring accordingly.